dot/network/message_test.go

```diff
@@ -207,14 +210,15 @@ func TestEncodeBlockResponseMessage_WithAll(t *testing.T) {
 
 func TestEncodeBlockAnnounceMessage(t *testing.T) {
 	// this value is a concatenation of:
+	//  SCALE option value: 0x01
 	//  ParentHash: Hash: 0x4545454545454545454545454545454545454545454545454545454545454545
 	//	Number: *big.Int // block number: 1
 	//	StateRoot:  Hash: 0xb3266de137d20a5d0ff3a6401eb57127525fd9b2693701f0bf5a8a853fa3ebe0
 	//	ExtrinsicsRoot: Hash: 0x03170a2e7597b7b7e3d84c05391d139a62b157e78786d8c082f29dcf4c111314
 	//	Digest: []byte
 
 	//                                    mtparenthash                                                      bnstateroot                                                       extrinsicsroot                                                di
-	expected, err := common.HexToBytes("0x454545454545454545454545454545454545454545454545454545454545454504b3266de137d20a5d0ff3a6401eb57127525fd9b2693701f0bf5a8a853fa3ebe003170a2e7597b7b7e3d84c05391d139a62b157e78786d8c082f29dcf4c1113140000")
+	expected, err := common.HexToBytes("0x01454545454545454545454545454545454545454545454545454545454545454504b3266de137d20a5d0ff3a6401eb57127525fd9b2693701f0bf5a8a853fa3ebe003170a2e7597b7b7e3d84c05391d139a62b157e78786d8c082f29dcf4c1113140000")
```

Comment:
why is this changed? it should be the same as before (in fact it must be, or interop will break)

Reply:
I believe it is because it was encoded as an optional so had a leading byte appended. Will look into working around this